### PR TITLE
Stop using s3 in editor

### LINKED ISSUE #1625: Unicode is not handled correctly by in-browser file editing
An example is `questions/derivativeUpload/question.html` in `pl-dev101`. It has a unicode `MINUS SIGN` in it (code point `0x2212`) which is displayed as the UTF-8 encoding of `0xE2 0x88 0x92` rather than the unicode character.

### PR DESCRIPTION
**Pull request by tbretl**
This PR fixes two bugs.

(1) The server-side base64 encoding and decoding was implemented incorrectly, and crashed the server when applied to text containing certain unicode characters (e.g., unicode minus). The implementation is now correct. It is also wrapped in a try/catch so if anything ever does go wrong, it won't crash the server.

(2) The first attempt at a fix introduced an even worse bug, as a consequence of forgetting that the AWS SDK reads and writes to S3 with a buffer and not a string. In production, this crashed the server on every attempt to edit a file. This second bug is also now fixed.

The reason the second bug wasn't caught by CI is that CI used local storage and not S3. To avoid this issue in future, this PR also switches from using S3 (in prod) and local storage (in dev) to using only `file-store.js` (in both prod and dev). This required a slight change to the `file-store` API.

For real this time, I hope:
Fixes #1625 and #1708

### CODE REVIEW

File: lib/base64-util.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    // (1) use encodeURIComponent to get percent-encoded UTF-8
+    // (2) convert percent encodings to raw bytes
+    // (3) convert raw bytes to Base64
+    try {
+        return btoa(encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, (match, p1) => {
+            return String.fromCharCode('0x' + p1);
+        }));
+    } catch(e) {
+        logger.error(`b64EncodeUnicode: failed to encode ${str}`);
+        return null;
```
Comment (mwest1066) [inline lib/base64-util.js]: Are callers prepared to handle `null`? Might be safer to return an empty string?

File: lib/file-store.js [hunk 2]
```diff
@@ -63,3 +66,27 @@ module.exports.delete = async (file_id, authn_user_id) => {
     await sqldb.callAsync('files_delete', params);
     debug('delete(): soft-deleted row in DB');
 };
+
+/**
+ * Get a file from the file store.
+ *
+ * @param {number} file_id - The file to get.
```
Comment (mwest1066) [inline lib/file-store.js]: Specify that the return value is a `Buffer` (or an object containing a Buffer and a file object)?

File: doc/installingNative.md [hunk 3]
```diff
@@ -88,6 +88,8 @@ npm run lint-js -s
 npm run lint-python -s
 ```
 
+* If you want to use the in-browser file editor while running natively, set `config.filesRoot` in `PrairieLearn/lib/config.js` to a convenient, existing directory. Temporary files will be saved there. Note that it is not necessary to set `config.filesRoot` to use `npm run test` - a separate directory will be assigned during testing.
```
[the remark this answers is not in the document]
Comment (tbretl) [reply doc/installingNative.md]: done

File: lib/file-store.js [hunk 4]
```diff
@@ -63,3 +66,27 @@ module.exports.delete = async (file_id, authn_user_id) => {
...
+    if (result.rows.length < 1) {
+        throw new Error(`No file with file_id ${file_id}`);
+    }
+
+    if (config.filesRoot == null) throw new Error(`config.filesRoot must be non-null to get file_id ${file_id} from file store`);
+    const filename = path.join(config.filesRoot, result.rows[0].storage_filename);
+
+    debug(`get(): readFile ${filename} and return buffer with contents`);
+    const contents = await fsPromises.readFile(filename);
+    return contents;
```
[the remark this answers is not in the document]
Comment (tbretl) [reply lib/file-store.js]: done

File: lib/base64-util.js (hunk 1 above)
Comment (tbretl) [reply lib/base64-util.js]: they are, but fair enough - empty string it is

File: lib/file-store.js (hunk 2 above)
Comment (tbretl) [reply lib/file-store.js]: done

**Pull request merged**